### PR TITLE
moving/removing consensus package mocks to testscommon

### PR DESCRIPTION
**Pull request by danielradu10**
## Reasoning behind the pull request
- 
- 
- 
  
## Proposed changes
- 
- 
- 

## Testing procedure
- 
- 
- 

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: consensus/mock/mockTestInitializer.go [hunk 1]
```diff
@@ -199,42 +201,42 @@ func InitConsensusCoreWithMultiSigner(multiSigner crypto.MultiSigner) *Consensus
...
-		headerSigVerifier:       headerSigVerifier,
-		fallbackHeaderValidator: fallbackHeaderValidator,
-		nodeRedundancyHandler:   nodeRedundancyHandler,
-		scheduledProcessor:      scheduledProcessor,
-		messageSigningHandler:   messageSigningHandler,
-		peerBlacklistHandler:    peerBlacklistHandler,
-		signingHandler:          signingHandler,
-		enableEpochsHandler:     enableEpochsHandler,
-	}
+	container := &consensusTestscommon.ConsensusCoreMock{}
```
[the remark this answers is not in the document]
Comment (danielradu10) [reply consensus/mock/mockTestInitializer.go:214]: should I keep it like this or change it?
also I think that moving the mockTestInitializer into the testscommon will lead to having the init as before without calling Set methods

File: consensus/broadcast/shardChainMessenger_test.go [hunk 2]
```diff
@@ -7,9 +7,12 @@ import (
 	"time"
 
 	"github.com/multiversx/mx-chain-core-go/data"
-	"github.com/multiversx/mx-chain-go/consensus"
 	"github.com/stretchr/testify/require"
 
+	"github.com/multiversx/mx-chain-go/consensus"
+	consensus2 "github.com/multiversx/mx-chain-go/testscommon/consensus"
```
Comment (ssd04) [inline consensus/broadcast/shardChainMessenger_test.go:13]: i suggest to rename this to `commonConsensus`, `testscommonConsensus` or something similar

File: consensus/spos/consensusMessageValidator_test.go [hunk 3]
```diff
@@ -7,16 +7,18 @@ import (
...
 	crypto "github.com/multiversx/mx-chain-crypto-go"
+	"github.com/stretchr/testify/assert"
+
 	"github.com/multiversx/mx-chain-go/common"
 	"github.com/multiversx/mx-chain-go/consensus"
 	"github.com/multiversx/mx-chain-go/consensus/mock"
 	"github.com/multiversx/mx-chain-go/consensus/spos"
 	"github.com/multiversx/mx-chain-go/consensus/spos/bls"
 	"github.com/multiversx/mx-chain-go/testscommon"
+	consensus2 "github.com/multiversx/mx-chain-go/testscommon/consensus"
```
Comment (ssd04) [inline consensus/spos/consensusMessageValidator_test.go:18]: suggestion to rename here

Comment (ssd04) [reply consensus/spos/consensusMessageValidator_test.go:18]: and in all other places

File: sharding/nodesCoordinator/indexHashedNodesCoordinatorWithRater_test.go [hunk 4]
```diff
@@ -21,7 +21,7 @@ import (
 	"github.com/multiversx/mx-chain-go/sharding/mock"
 	"github.com/multiversx/mx-chain-go/state"
 	"github.com/multiversx/mx-chain-go/testscommon/chainParameters"
-	"github.com/multiversx/mx-chain-go/testscommon/epochstartmock"
+	mock2 "github.com/multiversx/mx-chain-go/testscommon/epochstartmock"
```
Comment (ssd04) [inline sharding/nodesCoordinator/indexHashedNodesCoordinatorWithRater_test.go:24]: rename here

File: testscommon/consensus/mockTestInitializer.go [hunk 5]
```diff
@@ -7,13 +7,17 @@ import (
...
+
 	"github.com/multiversx/mx-chain-go/consensus"
+	"github.com/multiversx/mx-chain-go/consensus/mock"
 	"github.com/multiversx/mx-chain-go/sharding/nodesCoordinator"
 	"github.com/multiversx/mx-chain-go/testscommon"
-	consensusMocks "github.com/multiversx/mx-chain-go/testscommon/consensus"
+	"github.com/multiversx/mx-chain-go/testscommon/bootstrapperStubs"
 	"github.com/multiversx/mx-chain-go/testscommon/cryptoMocks"
 	"github.com/multiversx/mx-chain-go/testscommon/enableEpochsHandlerMock"
+	mock2 "github.com/multiversx/mx-chain-go/testscommon/epochstartmock"
```
Comment (ssd04) [inline testscommon/consensus/mockTestInitializer.go:18]: same here

**Comment by ssd04** [comment]
in consensus mock package, there is also `marshalizerMock` which can be used from testscommon, so it can be removed from here; there are also other mock related to singing which can be considered to be used from testscommon (or moved there)

File: testscommon/consensus/mockTestInitializer.go [hunk 6]
```diff
@@ -7,13 +7,17 @@ import (
...
+
 	"github.com/multiversx/mx-chain-go/consensus"
+	"github.com/multiversx/mx-chain-go/consensus/mock"
 	"github.com/multiversx/mx-chain-go/sharding/nodesCoordinator"
 	"github.com/multiversx/mx-chain-go/testscommon"
-	consensusMocks "github.com/multiversx/mx-chain-go/testscommon/consensus"
+	"github.com/multiversx/mx-chain-go/testscommon/bootstrapperStubs"
 	"github.com/multiversx/mx-chain-go/testscommon/cryptoMocks"
 	"github.com/multiversx/mx-chain-go/testscommon/enableEpochsHandlerMock"
+	testscommonConsensus "github.com/multiversx/mx-chain-go/testscommon/epochstartmock"
```
Comment (ssd04) [inline testscommon/consensus/mockTestInitializer.go:18]: this can be used directly as `epochstartmock` no need to label it

Comment (AdoAdoAdo) [inline testscommon/consensus/mockTestInitializer.go:18]: would leave the alias as epochstartMock

**Review by AdoAdoAdo — DISMISSED** [review]

**Review by ssd04 — APPROVED** [review]

**Review by ssd04 — APPROVED** [review]

**Review by AdoAdoAdo — APPROVED** [review]

**Pull request merged**